packages/utils/src/locking.js

```diff
@@ -0,0 +1,65 @@
+import path from 'path'
+import consola from 'consola'
+import hash from 'hash-sum'
+import fs from 'fs-extra'
+import properlock from 'proper-lockfile'
+import onExit from 'signal-exit'
+
+export const lockPaths = new Set()
+
+export const defaultLockOptions = {
+  stale: 30000,
```

Comment:
There is no direct need for checking the validity of a lockfile when a lock already exists, `proper-lockfile` sets a timer which updates the lockfile regularly and if a lockfile exists it already checks for staleness. See here for more info: https://github.com/moxystudio/node-proper-lockfile#comparison

I have set a default value for staleness of 30s, but maybe thats a bit much and we can just use the default value.